Userland/Libraries/LibCore/System.cpp

```diff
@@ -49,6 +50,15 @@ ErrorOr<void> sigaction(int signal, struct sigaction const* action, struct sigac
     return {};
 }
 
+ErrorOr<sighandler_t> signal(int signal, sighandler_t handler)
+{
+    struct sigaction old_action = {};
+    struct sigaction new_action = {};
+    new_action.sa_handler = handler;
+    TRY(Core::System::sigaction(signal, &new_action, &old_action));
+    return old_action.sa_handler;
```

Comment:
Ok, so it seems like all other syscall wrappers invoke their underlying syscall directly (rather than another wrapper) and return `Error::from_syscall` if it fails. Why not do the same here?

```c++
ErrorOr<sighandler_t> signal(int signum, sighandler_t handler)
{
    auto old_handler = ::signal(signum, handler);
    if (old_handler == SIG_ERR)
        return Error::from_syscall("signal"sv, -errno);
    return old_handler;
}
```

Otherwise, if `signal` does fail, the error message printed to the console by LibMain will say that sigaction is the call that failed, which may be a bit misleading.

Reply:
Also check the CI failure, it seems the definition for `sighandler_t` is missing on macOS. Is a #include missing?